rust/stub/src/uefi_helpers.rs

```diff
@@ -27,32 +25,36 @@ pub fn read_all(file: &mut RegularFile) -> Result<Vec<u8>> {
     Ok(buf)
 }
 
-/// Open the currently executing image as a file.
-pub fn booted_image_file(boot_services: &BootServices) -> Result<RegularFile> {
-    let mut file_system = boot_services.get_image_file_system(boot_services.image_handle())?;
+#[derive(Debug, Clone, Copy)]
+pub struct PeInMemory {
+    image_base: *const c_void,
+    image_size: usize,
+}
 
-    // The root directory of the volume where our binary lies.
-    let mut root = file_system.open_volume()?;
+impl PeInMemory {
+    /// Return a reference to the currently running image.
+    ///
+    /// # Safety
+    ///
+    /// The returned slice covers the whole loaded image in which we
+    /// currently execute. This means the safety guarantees of
+    /// [`core::slice::from_raw_parts`] that we use in this function
+    /// are only guaranteed, if we we don't mutate anything in this
+    /// range. This means no modification of global variables or
+    /// anything.
```

Comment:
Do you think we can tuck this into `uefi-rs` at some point?
I am fine with this `unsafe` block actually, this is fairly honest.

Reply:
What would you want to move into uefi-rs?

Reply:
This whole as slice memory API, I get it they cannot really offer better safety, but I guess they're closer to the problems this can cause than us? In that sense, they're better long term maintainers of this chunk of code. But that's an open discussion to have?